### ISSUE TITLE
Link of the localhost port not updated if we choose another port which is not 8888

### ISSUE THREAD

**Issue by mikibonacci** [aiidalab/aiidalab#436]
if we want to expose the (docker) aiidalab to the localhost port which is different from 8888 (the default), the link to which is printed in the log is not updated, and this may create confusion. If we run the container with, for example, -p 8887:8888, the link still tries to go to the local 8888 port

**Comment by danielhollas** [comment]
Which log exactly do you mean? Which link? How exactly are you running the container?

**Comment by mikibonacci** [comment]
We can also just update the documentation to explain this step. However, it is a corner-case as usually the user will use the 8888 port also for localhost

**Comment by danielhollas** [comment]
Yeah, but this is printed by the Jupyter from within the container, and it only knows the port that is connected inside, it can't know the exposed port on the outside. So that's not something that can be fixed.

By the way: Why would you run docker directly instead of `aiidalab-launch`, which handles this correctly? (it prints the proper URL).

**Comment by mikibonacci** [comment]
I usually do this if I want to change the exposed port to something different to 8888. I didn't know it was possible via `aiidalab-launch`. 
Also in the docs, we offer the possibility to use docker [directly](https://aiidalab.readthedocs.io/en/latest/usage/index.html#aiidalab-launch). 

Actually, the user used `aiidalab-launch` via docker desktop (Mac) and I think that after a first attempt he started a new container by putting the 8887 port in the optional settings via the docker desktop GUI.

**Comment by danielhollas** [comment]
You can change the port via profile configuration with `aiidalab-launch profile edit`. I'd really recommend using aiidalab-launch where possible, maybe we should make that point stronger.

**Comment by mikibonacci** [comment]
I agree

**Comment by danielhollas** [comment]
I believe this one can be closed based on the discussion above, @mikibonacci please let me know if you disagree.

**Comment by mikibonacci** [comment]
Hi @danielhollas , yes we can close this! Thanks for asking

**Comment by danielhollas** [comment]
@mikibonacci thanks. Also,can you please report the issue you had with aiidalab-launch and pipx?
